qiskit/providers/ibmq/managed/managedjobset.py

```diff
@@ -418,6 +419,42 @@ def name(self) -> str:
         """
         return self._name
 
+    def update_name(self, name: str) -> str:
+        """Update the name of this job set.
+
+        Args:
+            name: The new `name` for this job set.
+
+        Returns:
+            The new name associated with this job set.
+
+        Raises:
+            IBMQJobManagerInvalidStateError: If the input job name is not a string.
+            IBMQJobManagerUpdateError: If there was an error with updating the names
+                for the jobs in the job set.
+        """
+        if not isinstance(name, str):
+            raise IBMQJobManagerInvalidStateError(
+                '"{}" of type "{}" is not a valid job name. '
+                'The job name needs to be a string.'.format(name, type(str)))
+
+        for i, job in enumerate(self.jobs()):
+            if job:
+                try:
+                    _ = job.update_name("{}_{}_".format(name, i))
```

Comment:
Since the job name format (`{}_{}_`) is now used at multiple places, it's better to either declare it as a constant or put it in a method.